File: frontend/dev_vault/lib/data/models/resume_model.dart
class ResumeModel {
  final String id;
  final String fileName;
  final String fileUrl;
  final String fileSize;
  final String publicId;
  final DateTime uploadedAt;

  ResumeModel({
    required this.id,
    required this.fileName,
    required this.fileUrl,
    required this.fileSize,
    required this.publicId,
    required this.uploadedAt,
  });

  factory ResumeModel.fromJson(Map<String, dynamic> json) {
    return ResumeModel(
      id: json['_id'] as String? ?? json['id'] as String,
      fileName: json['fileName'] as String,
      fileUrl: json['fileUrl'] as String,
      fileSize: json['fileSize'] as String,
      publicId: json['publicId'] as String,
      uploadedAt: DateTime.parse(json['createdAt'] as String? ?? json['uploadedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fileName': fileName,
      'fileUrl': fileUrl,
      'fileSize': fileSize,
      'publicId': publicId,
    };
  }
}

File: backend/src/resume/resume.model.ts
import { timeStamp } from "console";
import mongoose from "mongoose";

export const ResumeSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "user",
      required: true,
    },
    fileName: {
      type: String,
      required: true,
      trim: true,
    },
    fileUrl: {
      type: String,
      required: true,
      trim: true,
    },
    fileSize: {
      type: String,
      required: true,
      trim: true,
    },
    publicId: {
      type: String,
      required: true,
      trim: true,
    },

    //        └─ Field: fileSize → type: String
    //    └─ Field: fileUrl → type: String
    //    └─ Field: publicId → type: String (for Cloudinary)
    //    └─ Don't forget: timestamps: true
  },
  {
    timestamps: true,
  },
);

 const ResumeModel = mongoose.model("/resume", ResumeSchema);
export default ResumeModel;